relay-server/src/actors/global_config.rs

```diff
@@ -146,8 +147,39 @@ pub struct GlobalConfigService {
 impl GlobalConfigService {
     /// Creates a new [`GlobalConfigService`].
     pub fn new(config: Arc<Config>, upstream: Addr<UpstreamRelay>) -> Self {
-        let (global_config_watch, _) = watch::channel(Arc::default());
         let (internal_tx, internal_rx) = mpsc::channel(1);
+
+        let (global_config_watch, _) = match (
+            config.relay_mode(),
+            config.has_credentials(),
+            config.global_config(),
+        ) {
+            (RelayMode::Proxy | RelayMode::Static, true, None) | (RelayMode::Managed, true, _) => {
+                relay_log::info!("global config service starting");
+                // This request will trigger the request intervals when internal_rx receives the
+                // result from upstream.
+                Self::request_global_config(upstream.clone(), internal_tx.clone());
+                watch::channel(Arc::new(GlobalConfig::default()))
+            }
+            (RelayMode::Proxy | RelayMode::Static, false, None)
+            | (RelayMode::Managed, false, _) => {
+                // NOTE(iker): not making a request results in the sleep handler
+                // not being reset, so no new requests are made.
+                relay_log::info!("global config service starting with fetching disabled: no credentials configured");
+                watch::channel(Arc::new(GlobalConfig::default()))
+            }
+            (RelayMode::Proxy | RelayMode::Static | RelayMode::Capture, _, Some(global_config)) => {
+                relay_log::info!("global config service starting with fetching disabled: using static global config");
+                watch::channel(global_config.clone())
+            }
+            (RelayMode::Capture, _, None) => {
+                relay_log::info!(
+                    "global config service starting with fetching disabled: using default config"
+                );
+                watch::channel(Arc::new(GlobalConfig::default()))
+            }
+        };
```

Comment:
looks a bit unwieldy but im not sure theres a better way given all the variables that affect the outcome. 

To any non-ikers reading this: the logic is from this specification: https://github.com/getsentry/team-ingest/issues/183#issuecomment-1705405809  
 
I moved this logic from spawn_handler because I didn't want to initialize the global_config_watch with default if we are supposed to use static file, although I suppose it might not matter.